### PR TITLE
RSDK-3514: parametrize gantry testlimit timeout 

### PR DESCRIPTION
**Pull request by martha-johnston**
This PR parametrizes gantry testlimit timeout based on `mm_per_rev`, `rpm`, and `length_mm`

### CODE REVIEW

File: components/gantry/singleaxis/singleaxis_test.go [hunk 1]
```diff
@@ -546,6 +546,39 @@ func TestTestLimit(t *testing.T) {
 	test.That(t, pos, test.ShouldEqual, float64(1))
 }
 
+func TestTestLimitTimeout(t *testing.T) {
```
Comment (randhid) [inline components/gantry/singleaxis/singleaxis_test.go:549]: Nice test!

File: components/gantry/singleaxis/singleaxis.go [hunk 2]
```diff
@@ -453,7 +453,11 @@ func (g *singleAxis) testLimit(ctx context.Context, pin int) (float64, error) {
 		}
 
 		elapsed := start.Sub(start)
-		if elapsed > (time.Second * 15) {
+		timeout := 15.0
```
Comment (randhid) [inline components/gantry/singleaxis/singleaxis.go:456]: [nit] constants at top of the file, and set to a default name to something like `homingTimeout` with a comment.

File: components/gantry/singleaxis/singleaxis.go [hunk 3]
```diff
@@ -453,7 +453,11 @@ func (g *singleAxis) testLimit(ctx context.Context, pin int) (float64, error) {
...
+		if g.mmPerRevolution != 0 && g.rpm != 0 && g.lengthMm != 0 {
```
Comment (randhid) [inline components/gantry/singleaxis/singleaxis.go:457]: comment add
```suggestion
		// if the parameters checked are non-zero, calculate a timeout with a safety factor of
		// 1.1 to complete the gantry's homing sequence to find the limit switches
		if g.mmPerRevolution != 0 && g.rpm != 0 && g.lengthMm != 0 {
```

**Review by randhid — APPROVED** [review]
LGTM! Few nits! Thank you for writing a test too!

**Comment by martha-johnston** [comment]
@randhid would you take one more look at the unit changes I just made? does that still look alright?

**Comment by martha-johnston** [comment]
yeah, using nanoseconds now because `time.Since` returns nanoseconds, so I found it easier to just calculate the timeout in nanoseconds. let me know if you'd prefer milliseconds tho, I think it will just be a little messier. just updated the comments to say nanoseconds as well

**Comment by randhid** [comment]
> yeah, using nanoseconds now because `time.Since` returns nanoseconds, so I found it easier to just calculate the timeout in nanoseconds. let me know if you'd prefer milliseconds tho, I think it will just be a little messier. just updated the comments to say nanoseconds as well

This way works and makes sense since we're relying on `time.Since` I think it should stay as is.

**Comment by martha-johnston** [comment]
sounds good. going to test on the soleng gantry real quick and then will merge

**Pull request merged**